### ISSUE TITLE
Failed to overfit during synchronization transformer training

### ISSUE THREAD

**Issue by a84379516** [v-iashin/Synchformer#5]
Hi!

Great work! I have tried overfitting during the synchronization stage on the validation data. I.E. train on validation set and evaluate on the fixed offsets for validation. However, the model does not seem to learn anything, the accuracy and loss having more or less the same values in the beginning of the training and after 35 epochs. I am running this experiment on VGGSound with the frozen pre-trained audio and visual encoders from the checkpoint below. Would you be able to advise about this? Thank you!

I have used this set of command lines:
S1_CKPT_ID="23-12-22T16-13-38"  # replace this with an exp folder name
EPOCH="best"

TRANSFORMERS_OFFLINE=1 torchrun --nproc_per_node=4 main.py \
    config=./configs/sync.yaml \
    logging.logdir=./logs/avclip_sync_models \
    data.vids_path=/home/VGGSound/h264_video_25fps_256side_16000hz_aac \
    data.dataset.target=dataset.vggsound.VGGSound \
    model.params.vfeat_extractor.params.ckpt_path="/home/Synchformer/logs/avclip_models/${S1_CKPT_ID}/epoch_${EPOCH}.pt" \
    model.params.afeat_extractor.params.ckpt_path="/home/Synchformer/logs/avclip_models/${S1_CKPT_ID}/epoch_${EPOCH}.pt" \
    training.base_batch_size=16

**Comment by v-iashin** [comment]
Hi, I noticed a similar behaviour when tried to overfit as small sample as one batch with 2 vids. If the offsets are sampled randomly, i.e. the default behaviour, the model does not learn anything, unlike tasks where inputs are mostly fixed.

I hypothesise that random offset picking provides strong augmentation to training, and it is difficult for the model to memorize all combinations of the inputs.

To overfit on your data, try fixing the offset for each video, like `v_start` and `offset_sec`, and train the model. The overfitting should happen quickly.

**Comment by a84379516** [comment]
Thanks, that worked indeed!
